### PR TITLE
Azure Pipelines

### PR DESCRIPTION
**Pull request by dpgraham**

### CODE REVIEW

**Comment by dpgraham** [comment]
@mykola-mokhnach It should be ready now

I had to fix the keyboard tests by adding a `click` so that the elements would be focussed on.

I also skipped the `mobile: scrollBackTo` test. That wasn't working on old ARM based emulators and still isn't working. Not sure how it ever worked because we're scrolling so that an element is out of the viewport and then scrolling back to it by using it's elementId.

**Comment by dpgraham** [comment]
I have a nightly build in this that's scheduled to run in the evening that runs across different SDK versions.... I want to be a little more conservative with how we use our concurrency so we don't always have to wait so long for pull requests.

File: ci-jobs/real-device-build.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+jobs:
```
Comment (dpgraham) [inline ci-jobs/real-device-build.yml]: This is a work-in-progress

**Comment by dpgraham** [comment]
@mykola-mokhnach this is ready now.

I had to skip the `mobile: scrollBackTo` test because I don't think it ever works. A couple of tests I skipped for old SDK versions on CI.

File: ci-jobs/ci.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    variables:
+      CI: true
+      TERM: dumb
+    steps:
+    - task: NodeTool@0
+      inputs:
+        versionSpec: 11.x
+    - script: npm install
+      displayName: Install Node dependencies
+    - script: npm test
```
Comment (mykola-mokhnach) [inline ci-jobs/ci.yml]: why do we need to execute npm test twice?

File: test/functional/commands/keyboard/keyboard-e2e-specs.js [hunk 3]
```diff
@@ -227,17 +228,20 @@ describe('keyboard', function () {
...
           let adb = new ADB();
-          if (parseInt(await adb.getApiLevel(), 10) < 24) {
+          let apiLevel = parseInt(await adb.getApiLevel(), 10);
+          if (apiLevel < 24 || (process.env.CI && apiLevel < 28)) {
             // below Android 7.0 (API level 24) typing too many characters in a
             // length-limited field will either throw a NullPointerException or
             // crash the app
-            return this.skip();
+            // also can be flakey in CI for SDK < 28
+            this.skip();
```
Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: why return has been removed?

**Pull request merged**